### PR TITLE
drivers: sensor: add tsl2561 basic support

### PR DESCRIPTION
**Pull request by ggrs**
Add basic support for ams TSL2561 light sensor. Triggers, attributes and manual integration time are currently not supported.

### CODE REVIEW

File: dts/bindings/sensor/ams,tsl2561.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    OSRAM ams TSL2561 light sensor.
+
+compatible: "ams,tsl2561"
+
+include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
+  integration-time:
+    type: int
+    default: 402
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ams,tsl2561.yaml:14]: Please explain why you chose this default in the description. Is it a register reset value?

File: dts/bindings/sensor/ams,tsl2561.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    description: |
+      ADC integration time in ms.
+    enum:
+      - 13
+      - 101
+      - 402
+  gain:
+    type: int
+    default: 16
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ams,tsl2561.yaml:23]: Please explain why you chose this default in the description. Is it a register reset value?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/bindings/sensor/ams,tsl2561.yaml (hunk 1 above)
Comment (ggrs) [reply dts/bindings/sensor/ams,tsl2561.yaml:14]: Yes, it is the register's reset value. The description was updated to include this information.

File: dts/bindings/sensor/ams,tsl2561.yaml (hunk 2 above)
Comment (ggrs) [reply dts/bindings/sensor/ams,tsl2561.yaml:23]: Yes, it is the register's reset value. The description was updated to include this information.

**Review by MaureenHelm — DISMISSED** [review]
Looks good, but needs a rebase

**Comment by ggrs** [comment]
@MaureenHelm rebase done

**Review by MaureenHelm — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**